File: utilities/convert_trace_into_wiggle.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <float.h>
#include <assert.h>

#include "../src/trace.h"

void usage()
{
    fprintf(stderr, "Usage: ./convert_trace_into_wiggle input.bin.trace [filter_threshold] > output.wig\n");
}

int 
main( int argc, char** argv )
{
    if( argc <= 1 || argc >= 4 )
    {
        usage();
        exit( -1 );
    }

    /* load the trace into memory */
    struct trace_t* trace;
    load_trace_from_file( &trace, argv[1] );

    double filter_threshold = 0;
    if( argc == 3 )
    {
        filter_threshold = atof( argv[2] );
    }
    
    write_wiggle_from_trace_to_stream( trace, stdout, filter_threshold );
    
    close_traces( trace );
    
    return 0;
}

File: utilities/convert_trace_into_wiggle.py
#!/usr/bin/python

import sys
import os

# add python_lib to sys.path (at the front, since there's a trace module in 
# the standard library )
sys.path.insert(0, "../python_lib/" )

from trace import *
from wiggle import *

def usage():
    print "Usage: ./convert_trace_into_wiggle input.bin.trace [filter_threshold] > output.wig"
    sys.exit(1)

def main():
    if len(sys.argv) not in (2, 3): usage()

    # load the trace into memory
    trace = load_c_trace_from_file( sys.argv[1] )

    filter_threshold = 0
    if len(sys.argv) == 3:
        filter_threshold = float( sys.argv[2] )

    write_wiggle_from_trace_to_stream( trace, sys.stdout, filter_threshold )

if __name__ == "__main__": main()
